python/src/trezorlib/cardano.py

```diff
@@ -240,28 +322,35 @@ def parse_certificate(certificate) -> CertificateWithPoolOwnersAndRelays:
 
     certificate_type = certificate["type"]
 
+    # TODO refactor - stake delegation and stake reg/dereg have too much common/duplicated logic
```

Comment:
Perhaps we should do it within this PR.

Reply:
Oh, I already tried and decided it wasn't worth it. I just forgot to remove the TODO.